### DISCUSSION TITLE
XAxis equidistantPreserveStart calculation

### DISCUSSION THREAD

**Discussion by mfialko** [recharts/recharts#5197] [Q&A] [answered]
I'm looking how to render bar chart labels using `equidistantPreserveStart` to evenly display only the labels for which there is enough space. If there isn't enough space, the label should be omitted to avoid overlap. How can this be achieved? Here's the current view of my chart:

<img width="199" alt="Screenshot 2024-11-04 at 00 57 12" src="https://github.com/user-attachments/assets/2d52318f-e86d-4259-ad15-6d3a6f6b1325">

Text is overflowing from the container. 
The same behaviour in the documentation:

<img width="1012" alt="Screenshot 2024-11-04 at 00 58 03" src="https://github.com/user-attachments/assets/15430b5c-eccd-43dc-bede-996a56720569">

Here is [Codesanbox](https://codesandbox.io/p/sandbox/line-chart-of-different-axis-intervals-forked-dx54zp)

Help me to fix overflowing of the text.
Don't show text at all if we don't have enough space or add three dots to text

**Comment by ckifer** [comment]
Hi, this is due to chart margin. If you increase your margin there will be enough room for the tick to show. Technically it "fits within the chart" so it shows, but its very long and at the moment the tick is not aware of the edge of the char so it goes outside the bounds of the chart and gets cut off.

To work around this you can do things like rotate your labels or use `tickFormatter` to add ellipses if entries are > than a certain length.

Here is a modified version of your sandbox https://codesandbox.io/p/sandbox/line-chart-of-different-axis-intervals-forked-dx54zp

**Reply by mfialko** [reply]
Thanks for the idea with `tickFormatter`, I’ll give it a try, but anyway it feels like workaround. In my case, I don’t have a predefined list of labels, and the items can vary in length. I thought the chart would calculate whether a label fits within the chart. Could it be that `equidistantPreserveStart` always shows the first element, regardless of whether it fits? 

Did you change something in the sandbox? I checked, but I couldn’t find any differences.

**Reply by mfialko** [reply]
<img width="261" alt="Screenshot 2024-11-04 at 11 44 50" src="https://github.com/user-attachments/assets/934b2090-9a74-4eb8-90c3-38c99a52a8d7">

It looks like a bug. Should I open an issue about this?

**Reply by ckifer** [reply, marked as answer]
Yep it's preserved on purpose and is not a bug - that's the "preserveStart" part. Some people want to see that first (or last or both) tick no matter what. What you want probably is just "equidistant" without preserving anything. I've been wanting to work on that but we need to release 3.0 first. 

Recharts has a lot of places where extra work needs done on the implementers side to make things look decent. This is one of those things - they're implemented features. 

And oops, I did change the sandbox. I'll find the right link when I get a moment

**Reply by mfialko** [reply]
`tickFormatter` works fine, thanks!

<img width="244" alt="Screenshot 2024-11-04 at 19 55 26" src="https://github.com/user-attachments/assets/5cc82ba4-3aff-4623-93b8-f469e633e69b">

**Reply by ckifer** [reply]
https://codesandbox.io/p/sandbox/line-chart-of-different-axis-intervals-forked-s57ssq
this should have the changes I made using angle as well.

Glad that worked!
